### PR TITLE
Completed Private Events app with fixes and RSpec testing

### PR DESCRIPTION
**Pull request by PolarisTLX**
App completed along with RSpec testing for the controllers, models and views. Ready for review @nidalaa.

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -76,19 +66,8 @@ def event_params
...
-      redirect_to login_url
-    end
-  end
-
-  def correct_user
-    @event = Event.find(params[:id])
-    @user = User.find(@event.host_id)
-    redirect_to(current_user) unless @user == current_user
-    # this sends a user to their own page if they try to access a different profile
+  def find_event
```
Comment (nidalaa) [inline app/controllers/events_controller.rb]: if you call this method simply `event`, you will be able to simply use it when needed, e.g. in update:

```
def update
     if event.update_attributes(event_params)
       flash[:success] = "Event updated"
       redirect_to event
     else
       render 'edit'
     end
   end
```

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -37,13 +33,12 @@ def create
...
-    @event = Event.find(params[:id])
+    find_event
   end
 
   # UPDATE request
   def update
-    @event = Event.find(params[:id])
+    find_event
     if @event.update_attributes(event_params)
       # handle successful update
```
Comment (nidalaa) [inline app/controllers/events_controller.rb]: comment to :fire:

File: app/models/event.rb [hunk 3]
```diff
@@ -1,19 +1,25 @@
 class Event < ApplicationRecord
-
   belongs_to :host, class_name: "User"
   has_many :invites, foreign_key: :attended_event_id, dependent: :destroy
   has_many :attendees, through: :invites
 
+  after_create :invite_host_automatically
   # default_scope -> { order(created_at: :asc) }
   #default_scope -> { order(date: :asc) }
```
Comment (nidalaa) [inline app/models/event.rb]: coments 🔥

File: app/controllers/events_controller.rb [hunk 4]
```diff
@@ -1,33 +1,29 @@
...
   # GET request
   def show
-    @user = current_user
     @event = Event.find(params[:id])
     @attendees = @event.attendees.where("invites.accepted = ?", true)
     @invited = @event.attendees.where("invites.accepted = ?", false)
     # need to store event to carry over to invite guests page:
     # need to store a cookie to do this
-    session[:event_id] = @event.id
+    # session[:event_id] = @event.id
```
Comment (nidalaa) [inline app/controllers/events_controller.rb]: comment 🔥

File: app/controllers/events_controller.rb [hunk 5]
```diff
@@ -1,33 +1,29 @@
...
   def new
-    @user = current_user
     @event = Event.new
   end
 
   # GET request
   def show
-    @user = current_user
     @event = Event.find(params[:id])
     @attendees = @event.attendees.where("invites.accepted = ?", true)
```
Comment (nidalaa) [inline app/controllers/events_controller.rb]: why not in scopes?

File: app/views/invites/new.html.erb [hunk 6]
```diff
@@ -18,6 +18,8 @@
         <% end %>
       </ul>
       <br/><br/>
+
+      <%= f.hidden_field(:event_id, value: @event.id) %>
```
Comment (nidalaa) [inline app/views/invites/new.html.erb]: Do you need this field? even_id should be in url `/events/:event_id/invites/new`

File: config/routes.rb [hunk 7]
```diff
@@ -11,7 +11,15 @@
...
 
-  resources :users
+  resources :users, only: [:new, :create, :show, :edit, :update, :index]
+  resources :users do
+    resources :invites, only: [:new, :create]
+  end
+
   resources :events
-  resources :invites
+  resources :event do
```
Comment (nidalaa) [inline config/routes.rb]: resources `events` would be more RESTful

File: spec/models/event_spec.rb [hunk 8]
```diff
@@ -1,5 +1,73 @@
 require 'rails_helper'
 
 RSpec.describe Event, type: :model do
-  # pending "add some examples to (or delete) #{__FILE__}"
+  before do
+    Rails.application.load_seed
```
Comment (nidalaa) [inline spec/models/event_spec.rb]: It's time to lear how to use FactoryBot :)
And please also read:  http://www.betterspecs.org/

**Review by nidalaa — CHANGES_REQUESTED** [review]

**Comment by nidalaa** [comment]
@PolarisTLX  @jklemon17  👆

**Comment by nidalaa** [comment]
@arielcamus could you review this app from aesthetic point of view? If it goes to portfolio, it should also look super cool, I guess;)

 I do not trust my own aesthetic taste :D

File: app/views/invites/new.html.erb (hunk 6 above)
Comment (jklemon17) [reply app/views/invites/new.html.erb]: The event_id is available in the url of the 'new' page, but does not seem to be available to the 'create' action when the form is submitted. The hidden field was the way we found to pass it to the create action.

**Comment by arielcamus** [comment]
Ok, so here are my overall comments about design:

TL;DR: Just copy other designs.

Long answer:

You guys should definitely join the Launch & Learn we will have in a couple of weeks with a former Airbnb designer. The talk is called "Design superpowers for developers".

The current design doesn't look modern right now. But don't worry... you are not supposed to be great designers. You are great developers already. I think the Lunch & Learn and some tips will help you a lot though.

Some things you should change for this project and all your designs in general are:

1. Don't use textures. Just use a white background.
2. Don't use manuscript fonts, stick to conventional fonts like Sans Serif or, if you want to use Google Fonts, Nunito.
3. Find a web that has a structure like yours and just copy their style. It's the best way to start acquiring a sense for good design. 
4. Design is not just how it looks, but more importantly, how it works. The user experience (UX). One important aspect of UX is reusing design patterns, in the same way you use patterns for code (e.g. MVC). In this case, I'd try to follow some common design patterns for the menus. Having Events, About and Contact on the right side is not something you see every day. Instead, you can just move those elements to the top bar. In the responsive version, you can hide them in the hamburger menu.

Now... all of this is a good amount of work. It's up to you to decide if you will do it or not. If you don't change the design, find a way to still add this project to your portfolio without showing the frontend. Focus on the code. How? By writing a really good README that drivers the employer/viewer attention to the code and the way it's structured.

Does that make sense?

File: app/views/invites/new.html.erb (hunk 6 above)
Comment (nidalaa) [reply app/views/invites/new.html.erb]: You can make it available:)

routes:

```
resources :events do
    resources :invites, only: [:new, :create]
  end

  resources :invites, only: [:update]
```

invites_controller:

```
 @event = Event.find(params[:event_id])
```

form:

```
 <%= form_for @invite, url: event_invites_path(@event) do |f| %>
```

**Comment by nidalaa** [comment]
@jklemon17 @PolarisTLX what has happend that you have a conflict?

File: app/views/invites/new.html.erb (hunk 6 above)
Comment (jklemon17) [reply app/views/invites/new.html.erb]: Thank you! Took me a few minutes to figure out that I had to change the tests, too, but it works.

**Comment by jklemon17** [comment]
Oops, we accidentally updated the README on the master branch instead of the dev branch (one of the pitfalls of updating the README directly on GitHub, instead of locally).
